src/Nethermind/Nethermind.Core/ExecutionRequest/ExecutionRequest.cs

```diff
@@ -0,0 +1,141 @@
+    {
+        int size = 0;
+        foreach (ExecutionRequest request in requests)
+        {
+            size += request.RequestData!.Length + 1;
+        }
+        return size;
+    }
+
+    public static void FlatEncodeWithoutType(this IEnumerable<ExecutionRequest> requests, Span<byte> buffer)
+    {
+        int currentPosition = 0;
+
+        foreach (ExecutionRequest request in requests)
+        {
+            // Ensure the buffer has enough space to accommodate the new data
+            if (currentPosition + request.RequestData!.Length > buffer.Length)
+            {
+                throw new InvalidOperationException("Buffer is not large enough to hold all data of requests");
+            }
+
+            // Copy the RequestData to the buffer at the current position
+            request.RequestData.CopyTo(buffer.Slice(currentPosition, request.RequestData.Length));
+            currentPosition += request.RequestData.Length;
+        }
+    }
+
+    public static ArrayPoolList<byte[]> GetFlatEncodedRequests(
+        IEnumerable<ExecutionRequest> depositRequests,
+        IEnumerable<ExecutionRequest> withdrawalRequests,
+        IEnumerable<ExecutionRequest> consolidationRequests
+    )
+    {
+        ArrayPoolList<byte[]> requests = new(3);
+        using ArrayPoolList<byte> depositBuffer = new(depositRequestsBytesSize);
+        using ArrayPoolList<byte> withdrawalBuffer = new(withdrawalRequestsBytesSize);
+        using ArrayPoolList<byte> consolidationBuffer = new(consolidationRequestsBytesSize);
+
+
+        foreach (ExecutionRequest request in depositRequests)
+        {
+            depositBuffer.AddRange(request.RequestData!);
+        }
+
+        foreach (ExecutionRequest request in withdrawalRequests)
+        {
+            withdrawalBuffer.AddRange(request.RequestData!);
+        }
+
+        foreach (ExecutionRequest request in consolidationRequests)
+        {
+            consolidationBuffer.AddRange(request.RequestData!);
+        }
+
+        requests.AddRange(depositBuffer.ToArray());
+        requests.AddRange(withdrawalBuffer.ToArray());
+        requests.AddRange(consolidationBuffer.ToArray());
+        return requests;
+    }
+
+    public static Hash256 CalculateHashFromFlatEncodedRequests(byte[][]? flatEncodedRequests)
+    {
+        // make sure that length is exactly 3
+        if (flatEncodedRequests is null || flatEncodedRequests.Length != 3)
+        {
+            throw new ArgumentException("Flat encoded requests must be an array of 3 elements");
+        }
+
+        using (SHA256 sha256 = SHA256.Create())
+        {
+            byte[] concatenatedHashes = new byte[32 * flatEncodedRequests!.Length];
+            int currentPosition = 0;
+            byte type = 0;
+            // Allocate the buffer once outside the loop
+            Span<byte> requestBuffer = stackalloc byte[flatEncodedRequests.Max(r => r.Length) + 1];
```

Comment:
You know which is the biggest request size, so maybe just make a const pointing to DepositRequestSize

Reply:
the biggest size is not fixed, as the the elements from the flatEncodedRequests is a concatenated bytes of several request objects.  i.e
```csharp
flatEncodedRequests[0] = [0x00 ++ deposit_request_1 ++ deposit_request_2 ++ ... ++ deposit_request_n]
flatEncodedRequests[1] = [0x01 ++ withdrawal_request_1 ++ withdrawal_request_2 ++ ... ++ withdrawal_request_n]
flatEncodedRequests[2] = [0x02++ consolidation_request_1 ++ consolidation_request_2 ++ ... ++ consolidation_request_n]
```